### ISSUE TITLE
Figure out exactly how many hops between client and end host

### ISSUE THREAD

**Issue by rpanah** [FangfanLi/ErateLibrary#1]

**Comment by arianniaki** [comment]
The following code should take care of that:
`import subprocess

hostname = "128.200.2.222"
traceroute = subprocess.Popen(["traceroute", '-w', '60',hostname],stdout=subprocess.PIPE, stderr=subprocess.STDOUT)

lines = []
for line in iter(traceroute.stdout.readline,""):
    print(line)
    lines.append(line)

final_line = (lines[len(lines)-1])
num_of_hops = final_line.split(" ")
print "number of hops are : "+ str(num_of_hops[1])
`

**Comment by arianniaki** [comment]
The following code should take care of it:
```python
import subprocess

hostname = "128.200.2.222"
traceroute = subprocess.Popen(["traceroute", '-w', '60',hostname],stdout=subprocess.PIPE, stderr=subprocess.STDOUT)

lines = []
for line in iter(traceroute.stdout.readline,""):
    print(line)
    lines.append(line)

final_line = (lines[len(lines)-1])
num_of_hops = final_line.split(" ")
print "number of hops are : "+ str(num_of_hops[1])
```

**Comment by rpanah** [comment]
No. This won't give you the output. Some traceroute implementations show packets that are beyond the destination. This will not work in those cases. You need to properly implement a traceroute-like function that increments the TTL on TCP `SYN` packets until it gets a `SYNACK`.
